File: src/main/java/me/miki/shindo/gui/modmenu/category/impl/shared/FilterChip.java
package me.miki.shindo.gui.modmenu.category.impl.shared;

import me.miki.shindo.utils.mouse.MouseUtils;

public class FilterChip {

    private float x;
    private float y;
    private float width;
    private float height;
    private final Runnable onClick;

    public FilterChip(Runnable onClick) {
        this.onClick = onClick;
    }

    public void setBounds(float x, float y, float width, float height) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }

    public boolean contains(int mx, int my) {
        return MouseUtils.isInside(mx, my, x, y, width, height);
    }

    public void click() {
        if (onClick != null) {
            onClick.run();
        }
    }
}

File: src/main/kotlin/me/miki/shindo/ui/comp/chips/FilterChip.kt
package me.miki.shindo.ui.comp.chips

import me.miki.shindo.utils.mouse.MouseUtils

class FilterChip(private val onClick: Runnable?) {
    private var x = 0f
    private var y = 0f
    private var width = 0f
    private var height = 0f

    fun setBounds(x: Float, y: Float, width: Float, height: Float) {
        this.x = x
        this.y = y
        this.width = width
        this.height = height
    }

    fun contains(mx: Int, my: Int): Boolean {
        return MouseUtils.isInside(mx, my, x, y, width, height)
    }

    fun click() {
        onClick?.run()
    }
}
